Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -99,47 +121,95 @@ class LoginDetailViewController: UITableViewController {
+        // draws it's own seperators. The last item in the list draws its seperator full width. The reason
+
+        // Prevent seperators from showing by pushing them off screen by the width of the cell
+        let itemsToHideSeperators: [InfoItem] = [.WebsiteItem, .LastModifiedSeperator]
+        itemsToHideSeperators.forEach { item in
+            let cell = tableView.cellForRowAtIndexPath(NSIndexPath(forRow: item.rawValue, inSection: 0))
+            cell?.separatorInset = UIEdgeInsetsMake(0, cell?.bounds.width ?? 0, 0, 0)
+        }
+
+        // Rows to display full width seperator
+        let itemsToShowFullWidthSeperator: [InfoItem] = [.DeleteItem]
+        itemsToShowFullWidthSeperator.forEach { item in
+            let cell = tableView.cellForRowAtIndexPath(NSIndexPath(forRow: item.rawValue, inSection: 0))
+            cell?.separatorInset = UIEdgeInsetsZero
+            cell?.layoutMargins = UIEdgeInsetsZero
+            cell?.preservesSuperviewLayoutMargins = false
+        }
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension LoginDetailViewController: UITableViewDataSource {
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        switch InfoItem(rawValue: indexPath.row)! {
+        case .TitleItem:
+            let loginCell = dequeueLoginCellForIndexPath(indexPath)
+            loginCell.style = .IconAndDescriptionLabel
+            loginCell.descriptionLabel.text = login.hostname
             return loginCell
-        case .Delete:
-            let deleteCell = tableView.dequeueReusableCellWithIdentifier(DeleteCellIdentifier, forIndexPath: indexPath)
+
+        case .UsernameItem:
+            let loginCell = dequeueLoginCellForIndexPath(indexPath)
+            loginCell.style = .NoIconAndBothLabels
+            loginCell.highlightedLabel.text = NSLocalizedString("username", tableName: "LoginManager", comment: "Title for username row in Login Detail View")
+            loginCell.descriptionLabel.text = login.username
+            loginCell.descriptionLabel.keyboardType = .EmailAddress
+            loginCell.descriptionLabel.returnKeyType = .Next
+            loginCell.descriptionLabel.userInteractionEnabled = editingInfo
+            return loginCell
+
+        case .PasswordItem:
+            let loginCell = dequeueLoginCellForIndexPath(indexPath)
+            loginCell.style = .NoIconAndBothLabels
+            loginCell.highlightedLabel.text = NSLocalizedString("password", tableName: "LoginManager", comment: "Title for password row in Login Detail View")
+            loginCell.descriptionLabel.text = login.password
+            loginCell.descriptionLabel.returnKeyType = .Next
+            loginCell.displayDescriptionAsPassword = true
+            loginCell.descriptionLabel.userInteractionEnabled = editingInfo
+            return loginCell
+
+        case .WebsiteItem:
```

Comment:
How useful is it to edit the website a login is associated with? I definitely understand usernames and passwords in case they change or are mistyped, but I can't figure out any scenario where I'd want to change the website as a user.